src/components/button.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import Link from "next/link";
+import React, { useRef } from "react";
+
+interface Props {
+  url?: string;
+  clickFn?: () => void;
+  children: React.ReactNode;
+  btnclass: string;
+  type?: "button" | "submit" | "reset";
+}
+
+const Button = ({ url, clickFn, children, btnclass, type }: Props) => {
+  let buttonRef = useRef<HTMLButtonElement>(null);
+  let linkRef = useRef<HTMLAnchorElement>(null);
+
+  const component = url ? (
+    <Link href={url} passHref legacyBehavior>
+      <a ref={linkRef} className={btnclass}>
+        {children}
+      </a>
+    </Link>
```

Comment:
just regarding the link comment, I was trying to understand why you are using legacy behavior but the docs are not very clear so I wasn't sure of what's correct either, but I have tried to create something like the below and it worked for me so can you just clarify so I can learn what's happening here?

`'use client';
import Link from 'next/link';
import { useRef } from 'react';

type BtnProps = {
  name: string;
  className?: string;
};

const Btn = ({ name, className }: BtnProps) => {
  const ref = useRef(null);

  return (
    <>
      <Link href='/login' className={className} ref={ref}>
        {name}
      </Link>
    </>
  );
};

export default Btn;`

Anyway I'll accept your solution, just want to understand why.

Reply:
you can pass reff to a link so the only way suggested is to have an a tag within the link. however that is apparently deprecated as of nextjs 10 I think so you need to use legacy in order to enable it

Reply:
do you have a source for this as I couldn't find in their comments and I tested the file attached where I pass the ref directly. I will log the ref later today to see if I'm missing something